LiteCore/Query/SQLiteDataFile+Indexes.cc

```diff
@@ -122,26 +125,68 @@ namespace litecore {
         unregisterIndex(spec.name);
         if ( spec.type != IndexSpec::kFullText && spec.type != IndexSpec::kVector )
             exec(CONCAT("DROP INDEX IF EXISTS " << sqlIdentifier(spec.name)));
-        if ( !spec.indexTableName.empty() ) garbageCollectIndexTable(spec.indexTableName);
+        if ( !spec.indexTableName.empty() ) garbageCollectIndexTable(spec);
     }
 
     // Drops unnested-array tables that no longer have any indexes on them.
-    void SQLiteDataFile::garbageCollectIndexTable(const string& tableName) {
-        {
-            SQLite::Statement stmt(*this, "SELECT name FROM indexes WHERE indexTableName=?");
-            stmt.bind(1, tableName);
-            if ( stmt.executeStep() ) return;
+    void SQLiteDataFile::garbageCollectIndexTable(const SQLiteIndexSpec& spec) {
+        string              tableName = spec.indexTableName;  // tableName is in plain form.
+        std::vector<string> unnestTables;
+        if ( spec.type == IndexSpec::kArray ) {
+            for ( size_t pos = 0; pos != string::npos; ) {
+                pos = tableName.find(KeyStore::kUnnestLevelSeparator, pos);
+                if ( pos != string::npos ) {
+                    unnestTables.push_back(tableName.substr(0, pos));
+                    pos += KeyStore::kUnnestLevelSeparator.size;
+                } else {
+                    unnestTables.push_back(tableName);
+                }
+            }
+            // Assertion: unnestTables.back() == tableName
```

Comment:
Use DebugAssert to specify assertions/invariants.

Reply:
Okay